include/graaflib/algorithm/shortest_path.tpp

```diff
@@ -153,4 +153,45 @@ dijkstra_shortest_paths(const graph<V, E, T>& graph,
   return shortest_paths;
 }
 
+template <typename V, typename E, graph_type T, typename WEIGHT_T>
+std::unordered_map<vertex_id_t, graph_path<WEIGHT_T>>
+bellman_ford_shortest_paths(const graph<V, E, T>& graph, vertex_id_t start_vertex) {
+
+  std::unordered_map<vertex_id_t, graph_path<WEIGHT_T>> shortest_paths;
+
+  // Initialize the shortest path distances from the starting vertex to "infinity" for all vertices
+  for (const auto& vertex : graph.get_vertices()) {
+    shortest_paths[vertex.first] = {
+      {vertex.first},
+      std::numeric_limits<WEIGHT_T>::max()
+    };
+  }
```

Comment:
[very minor]: You could use structured binding here to get rid of the `pair` type. This makes the variable names slightly more readable:
```suggestion
  for (const auto& [vertex_id, _] : graph.get_vertices()) {
    shortest_paths[vertex_id] = {
      {vertex_id},
      std::numeric_limits<WEIGHT_T>::max()
    };
  }
```